lib/src/main/java/uk/gov/di/ipv/cri/passport/library/validation/JarValidator.java

```diff
@@ -0,0 +1,187 @@
+package uk.gov.di.ipv.cri.passport.library.validation;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWEObject;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.crypto.ECDSAVerifier;
+import com.nimbusds.jwt.JWTClaimNames;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import com.nimbusds.jwt.proc.BadJWTException;
+import com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier;
+import com.nimbusds.oauth2.sdk.OAuth2Error;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.ssm.model.ParameterNotFoundException;
+import uk.gov.di.ipv.cri.passport.library.exceptions.SharedAttributesValidationException;
+import uk.gov.di.ipv.cri.passport.library.helpers.JwtHelper;
+import uk.gov.di.ipv.cri.passport.library.service.ConfigurationService;
+
+import java.net.URI;
+import java.text.ParseException;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+
+public class JarValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JarValidator.class);
+    private static final String REDIRECT_URI_CLAIM = "redirect_uri";
+
+    private final ConfigurationService configurationService;
+
+    public JarValidator(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    public SignedJWT decryptJWE(String jweString) throws SharedAttributesValidationException {
+        try {
+            JWEObject jweObject = JWEObject.parse(jweString);
+
+            // Decrypt functionality to go here...
+
+            return jweObject.getPayload().toSignedJWT();
+        } catch (ParseException e) {
+            LOGGER.error("Failed to parse request body into a JWE");
+            throw new SharedAttributesValidationException(OAuth2Error.INVALID_REQUEST_OBJECT);
+        }
+    }
+
+    public JWTClaimsSet validateRequestJwt(SignedJWT signedJWT, String clientId)
+            throws SharedAttributesValidationException {
+        validateClientId(clientId);
+        validateJWTHeader(signedJWT, clientId);
+        validateSignature(signedJWT, clientId);
+        JWTClaimsSet claimsSet = validateClaimSet(signedJWT, clientId);
+        validateRedirectUri(claimsSet, clientId);
+
+        return claimsSet;
+    }
+
+    private void validateClientId(String clientId) throws SharedAttributesValidationException {
+        try {
+            configurationService.getClientAuthenticationMethod(clientId);
+        } catch (ParameterNotFoundException e) {
+            LOGGER.error("Unknown client id provided {}", clientId);
+            throw new SharedAttributesValidationException(
```

Comment:
I think the scope of this Class is beyond just `SharedAttributes` now, so I would be inclined to a create new exception something like `JarValidationException`, similar comment for other places where `SharedAttributesValidationException` is being thrown within this class.

Reply:
Yep good idea, I'll change this.